.github/workflows/main.yml

```diff
@@ -0,0 +1,88 @@
+name: Docker build and push
+
+on:
+  push:
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+env:
+  # TEST_TARGET: Name of the testing target in the Dockerfile
+  TEST_TARGET: testing
+
+  # DO_TEST - true to build and run unit tests, false to skip the tests
+  DO_TEST: false
+
+  # DO_PUSH - true to push to the HPE_DEPLOY_REPO, false to not push
+  DO_PUSH: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Lowercase repository name for docker build
+      id: lowercase-repository-name
+      run: |
+        echo "HPE_BUILD_REPO=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
+        echo "HPE_DEPLOY_REPO=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
+
+    - name: set TAG & TESTTAG for main/master
+      if: ${{ github.event.repository.default_branch == 'main' || github.event.repository.default_branch == 'master' }}
+      run: |
+        echo "TAG=${{ github.sha }}" >> ${GITHUB_ENV}
+        echo "LATESTTAG=latest" >> ${GITHUB_ENV}
+        echo "TESTTAG=test-${{ github.sha }}" >> ${GITHUB_ENV}
```

Comment:
similar here, see if we can clean up the "TESTTAG" code throughout this workflow